src/sdkRuntimeModels.ts

```diff
@@ -217,14 +200,49 @@ export interface MParticleWebSDK {
         eventOptions?: SDKEventOptions
     ): void;
     logBaseEvent(event: BaseEvent, eventOptions?: SDKEventOptions): void;
+    logError(error: IErrorLogMessage, attrs?: SDKEventAttrs): void;
+    logLink(selector: string, eventName: string, eventType: valueof<typeof EventType>, eventInfo: SDKEventAttrs): void;
+    logForm(selector: string, eventName: string, eventType: valueof<typeof EventType>, eventInfo: SDKEventAttrs): void;
+    logPageView(eventName: string, attrs?: SDKEventAttrs, customFlags?: SDKEventCustomFlags, eventOptions?: SDKEventOptions): void;
+    setOptOut(isOptingOut: boolean): void;
     eCommerce: SDKECommerceAPI;
-    logLevel: string;
-    ProductActionType: SDKProductActionType;
+    isInitialized(): boolean;
+    ProductActionType: valueof<typeof ProductActionType>;
+    ready(f: Function): void;
+    reset(instance: IMParticleWebSDKInstance): void;
+    setAppName(name: string): void;
+    setAppVersion(version: string): void;
+    setOptOut(isOptingOut: boolean): void;
+    startTrackingLocation(callback?: Callback): void;
+    stopTrackingLocation(): void;
     generateHash(value: string): string;
+    setIntegrationAttribute(integrationModuleId: number, attrs: IntegrationAttribute): void;
+    getIntegrationAttributes(integrationModuleId: number): IntegrationAttribute;
```

Comment:
I think this should be `IntegrationAttributes`

Reply:
Not sure about that. Looking at the actual implementation, it returns a single `IntegrationAttribute`:

```
    this.getIntegrationAttributes = function(integrationId) {
        if (self._Store.integrationAttributes[integrationId]) {
            return self._Store.integrationAttributes[integrationId];
        } else {
            return {};
        }
    };
```

Reply:
[Setting it sets an object](https://github.com/mparticle-integrations/mparticle-javascript-integration-adobe/blob/65cc56d8c1a2caa0a2fc412eefd93b7073211fbd/packages/AdobeServer/src/AdobeServerSideKit.js#L90-L94).  Note the `else` above, it is returning an empty object, as opposed to a `null` if it were just returning a single item.

Reply:
Per our conversation, I am more correct than you. 😉